components/Chart/index.tsx

```diff
@@ -0,0 +1,260 @@
+import LineChart from '@/components/Chart/LineChart'
+import { bisect, curveCardinal, NumberValue, scaleLinear, timeDay, timeHour, timeMinute, timeMonth } from 'd3'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { AxisBottom, TickFormatter } from '@visx/axis'
+import {
+  dayHourFormatter,
+  hourFormatter,
+  monthDayFormatter,
+  monthTickFormatter,
+  monthYearDayFormatter,
+  weekFormatter,
+} from 'util/formatChartTimes'
+import { Line } from '@visx/shape'
+import { GlyphCircle } from '@visx/glyph'
+import { localPoint } from '@visx/event'
+import { EventType } from '@visx/event/lib/types'
+import { Color } from '@/const/styles/variables'
+import { getPriceChangeColor } from 'util/getPriceChangeColor'
+
+export type PricePoint = { timestamp: number; value: number }
+
+export enum TimePeriod {
+  HOUR,
+  DAY,
+  WEEK,
+  MONTH,
+  YEAR,
+}
```

Comment:
I didn't see this in the demo. This is meant to have in the future some control for this?